### PR TITLE
Fix magic-numbers checker

### PR DESCRIPTION
**Pull request by finomen**
magic-numbers produced false-positives on compiler-generated code for defaulted member functions

### CODE REVIEW

**Comment by Javier-varez** [comment]
Please, mention in the commit description what was broken and how this fixes it

File: clang-tools-extra/clang-tidy/readability/MagicNumbersCheck.h [hunk 1]
```diff
@@ -78,10 +78,6 @@ class MagicNumbersCheck : public ClangTidyCheck {
         CharSourceRange::getTokenRange(MatchedLiteral->getSourceRange()),
         *Result.SourceManager, getLangOpts());
 
-    const auto parents = Result.Context->getParents(*MatchedLiteral);
-
-    const auto val = parents[0].template get<CallExpr>();
-
```
Comment (Javier-varez) [inline clang-tools-extra/clang-tidy/readability/MagicNumbersCheck.h:84]: Was this deadcode? seems unrelated to this change

Comment (finomen) [reply clang-tools-extra/clang-tidy/readability/MagicNumbersCheck.h:84]: Yep, it's deadcode added in one of earlier revisions

**Review by Javier-varez — APPROVED** [review]
